From 3b813bb39f918eb9209a66a6ad2cc49ca2b6d6e9 Mon Sep 17 00:00:00 2001
From: TheGamerKing561 <eyadtamer2121@gmail.com>
Date: Sat, 11 Jul 2026 18:03:46 +0300
Subject: [PATCH] Add TW_REMOVE_BATTERY_TEXT flag

This flag removes the "Battery: " text from the status bar battery indicator when its set to true.

This flag is useful in cases where we need extra space to move one of the status bar items (like the clock) to be beside the battery percentage.
For example, on Xiaomi Mi A2 Lite (daisy), where we have a big notch in center that completely hides the clock in the status bar, and there is no space to move it beside the battery indicator due to the "Battery: " text.

Change-Id: If2d47b3a638eca358e0c961e1ba8dbb196468aa6
Signed-off-by: TheGamerKing561 <eyadtamer2121@gmail.com>
---

diff --git a/Android.mk b/Android.mk
index 383adca..1c8f81d 100644
--- a/Android.mk
+++ b/Android.mk
@@ -269,6 +269,9 @@
 ifeq ($(TW_NO_BATT_PERCENT), true)
     LOCAL_CFLAGS += -DTW_NO_BATT_PERCENT
 endif
+ifeq ($(TW_REMOVE_BATTERY_TEXT), true)
+    LOCAL_CFLAGS += -DTW_REMOVE_BATTERY_TEXT
+endif
 ifeq ($(TW_NO_CPU_TEMP), true)
     LOCAL_CFLAGS += -DTW_NO_CPU_TEMP
 endif
diff --git a/data.cpp b/data.cpp
index 0078524..7da3ea5 100644
--- a/data.cpp
+++ b/data.cpp
@@ -647,6 +647,12 @@
 #else
 	mConst.SetValue(TW_NO_BATTERY_PERCENT, "0");
 #endif
+#ifdef TW_REMOVE_BATTERY_TEXT
+    printf("TW_REMOVE_BATTERY_TEXT := true\n");
+    mConst.SetValue("tw_remove_battery_text", "1");
+#else
+    mConst.SetValue("tw_remove_battery_text", "0");
+#endif
 #ifdef TW_NO_CPU_TEMP
 	printf("TW_NO_CPU_TEMP := true\n");
 	mConst.SetValue("tw_no_cpu_temp", "1");
diff --git a/gui/theme/landscape_hdpi/ui.xml b/gui/theme/landscape_hdpi/ui.xml
index a71387f..ecc7b7f 100755
--- a/gui/theme/landscape_hdpi/ui.xml
+++ b/gui/theme/landscape_hdpi/ui.xml
@@ -302,6 +302,20 @@
 					<condition var1="tw_battery" op="&gt;" var2="0"/>
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 					<condition var1="tw_battery_pos_x" var2="0"/>
+					<condition var1="tw_remove_battery_text" var2="1"/>
+				</conditions>
+				<font resource="font_m"/>
+				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<text>%tw_battery%</text>
+			</text>
+
+			<text color="%text_color%">
+				<conditions>
+					<condition var1="tw_no_battery_percent" var2="0"/>
+					<condition var1="tw_battery" op="&gt;" var2="0"/>
+					<condition var1="tw_battery" op="&lt;" var2="101"/>
+					<condition var1="tw_battery_pos_x" var2="0"/>
+					<condition var1="tw_remove_battery_text" op="!=" var2="1"/>
 				</conditions>
 				<font resource="font_m"/>
 				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
@@ -348,6 +362,20 @@
 					<condition var1="tw_battery" op="&gt;" var2="0"/>
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 					<condition var1="tw_battery_pos_x" op="!=" var2="0"/>
+					<condition var1="tw_remove_battery_text" var2="1"/>
+				</conditions>
+				<font resource="font_m"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
+				<text>%tw_battery%</text>
+			</text>
+
+			<text color="%text_color%">
+				<conditions>
+					<condition var1="tw_no_battery_percent" var2="0"/>
+					<condition var1="tw_battery" op="&gt;" var2="0"/>
+					<condition var1="tw_battery" op="&lt;" var2="101"/>
+					<condition var1="tw_battery_pos_x" op="!=" var2="0"/>
+					<condition var1="tw_remove_battery_text" op="!=" var2="1"/>
 				</conditions>
 				<font resource="font_m"/>
 				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
diff --git a/gui/theme/landscape_mdpi/ui.xml b/gui/theme/landscape_mdpi/ui.xml
index dadfbb9..8025227 100755
--- a/gui/theme/landscape_mdpi/ui.xml
+++ b/gui/theme/landscape_mdpi/ui.xml
@@ -302,6 +302,20 @@
 					<condition var1="tw_battery" op="&gt;" var2="0"/>
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 					<condition var1="tw_battery_pos_x" var2="0"/>
+					<condition var1="tw_remove_battery_text" var2="1"/>
+				</conditions>
+				<font resource="font_m"/>
+				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<text>%tw_battery%</text>
+			</text>
+
+			<text color="%text_color%">
+				<conditions>
+					<condition var1="tw_no_battery_percent" var2="0"/>
+					<condition var1="tw_battery" op="&gt;" var2="0"/>
+					<condition var1="tw_battery" op="&lt;" var2="101"/>
+					<condition var1="tw_battery_pos_x" var2="0"/>
+					<condition var1="tw_remove_battery_text" op="!=" var2="1"/>
 				</conditions>
 				<font resource="font_m"/>
 				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
@@ -348,6 +362,20 @@
 					<condition var1="tw_battery" op="&gt;" var2="0"/>
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 					<condition var1="tw_battery_pos_x" op="!=" var2="0"/>
+					<condition var1="tw_remove_battery_text" var2="1"/>
+				</conditions>
+				<font resource="font_m"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
+				<text>%tw_battery%</text>
+			</text>
+
+			<text color="%text_color%">
+				<conditions>
+					<condition var1="tw_no_battery_percent" var2="0"/>
+					<condition var1="tw_battery" op="&gt;" var2="0"/>
+					<condition var1="tw_battery" op="&lt;" var2="101"/>
+					<condition var1="tw_battery_pos_x" op="!=" var2="0"/>
+					<condition var1="tw_remove_battery_text" op="!=" var2="1"/>
 				</conditions>
 				<font resource="font_m"/>
 				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
diff --git a/gui/theme/portrait_hdpi/ui.xml b/gui/theme/portrait_hdpi/ui.xml
index 104c3f6..918925c 100755
--- a/gui/theme/portrait_hdpi/ui.xml
+++ b/gui/theme/portrait_hdpi/ui.xml
@@ -299,6 +299,20 @@
 					<condition var1="tw_battery" op="&gt;" var2="0"/>
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 					<condition var1="tw_battery_pos_x" var2="0"/>
+					<condition var1="tw_remove_battery_text" var2="1"/>
+				</conditions>
+				<font resource="font_m"/>
+				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<text>%tw_battery%</text>
+			</text>
+
+			<text color="%text_color%">
+				<conditions>
+					<condition var1="tw_no_battery_percent" var2="0"/>
+					<condition var1="tw_battery" op="&gt;" var2="0"/>
+					<condition var1="tw_battery" op="&lt;" var2="101"/>
+					<condition var1="tw_battery_pos_x" var2="0"/>
+					<condition var1="tw_remove_battery_text" op="!=" var2="1"/>
 				</conditions>
 				<font resource="font_m"/>
 				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
@@ -345,6 +359,20 @@
 					<condition var1="tw_battery" op="&gt;" var2="0"/>
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 					<condition var1="tw_battery_pos_x" op="!=" var2="0"/>
+					<condition var1="tw_remove_battery_text" var2="1"/>
+				</conditions>
+				<font resource="font_m"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
+				<text>%tw_battery%</text>
+			</text>
+
+			<text color="%text_color%">
+				<conditions>
+					<condition var1="tw_no_battery_percent" var2="0"/>
+					<condition var1="tw_battery" op="&gt;" var2="0"/>
+					<condition var1="tw_battery" op="&lt;" var2="101"/>
+					<condition var1="tw_battery_pos_x" op="!=" var2="0"/>
+					<condition var1="tw_remove_battery_text" op="!=" var2="1"/>
 				</conditions>
 				<font resource="font_m"/>
 				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
diff --git a/gui/theme/portrait_mdpi/ui.xml b/gui/theme/portrait_mdpi/ui.xml
index d81a9a9..a404871 100755
--- a/gui/theme/portrait_mdpi/ui.xml
+++ b/gui/theme/portrait_mdpi/ui.xml
@@ -299,6 +299,20 @@
 					<condition var1="tw_battery" op="&gt;" var2="0"/>
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 					<condition var1="tw_battery_pos_x" var2="0"/>
+					<condition var1="tw_remove_battery_text" var2="1"/>
+				</conditions>
+				<font resource="font_m"/>
+				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<text>%tw_battery%</text>
+			</text>
+
+			<text color="%text_color%">
+				<conditions>
+					<condition var1="tw_no_battery_percent" var2="0"/>
+					<condition var1="tw_battery" op="&gt;" var2="0"/>
+					<condition var1="tw_battery" op="&lt;" var2="101"/>
+					<condition var1="tw_battery_pos_x" var2="0"/>
+					<condition var1="tw_remove_battery_text" op="!=" var2="1"/>
 				</conditions>
 				<font resource="font_m"/>
 				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
@@ -345,6 +359,20 @@
 					<condition var1="tw_battery" op="&gt;" var2="0"/>
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 					<condition var1="tw_battery_pos_x" op="!=" var2="0"/>
+					<condition var1="tw_remove_battery_text" var2="1"/>
+				</conditions>
+				<font resource="font_m"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
+				<text>%tw_battery%</text>
+			</text>
+
+			<text color="%text_color%">
+				<conditions>
+					<condition var1="tw_no_battery_percent" var2="0"/>
+					<condition var1="tw_battery" op="&gt;" var2="0"/>
+					<condition var1="tw_battery" op="&lt;" var2="101"/>
+					<condition var1="tw_battery_pos_x" op="!=" var2="0"/>
+					<condition var1="tw_remove_battery_text" op="!=" var2="1"/>
 				</conditions>
 				<font resource="font_m"/>
 				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
diff --git a/gui/theme/watch_mdpi/ui.xml b/gui/theme/watch_mdpi/ui.xml
index 8396639..37f86b8 100644
--- a/gui/theme/watch_mdpi/ui.xml
+++ b/gui/theme/watch_mdpi/ui.xml
@@ -238,6 +238,20 @@
 					<condition var1="tw_battery" op="&gt;" var2="0"/>
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 					<condition var1="tw_battery_pos_x" var2="0"/>
+					<condition var1="tw_remove_battery_text" var2="1"/>
+				</conditions>
+				<font resource="font_m"/>
+				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<text>%tw_battery%</text>
+			</text>
+
+			<text color="%text_color%">
+				<conditions>
+					<condition var1="tw_no_battery_percent" var2="0"/>
+					<condition var1="tw_battery" op="&gt;" var2="0"/>
+					<condition var1="tw_battery" op="&lt;" var2="101"/>
+					<condition var1="tw_battery_pos_x" var2="0"/>
+					<condition var1="tw_remove_battery_text" op="!=" var2="1"/>
 				</conditions>
 				<font resource="font_m"/>
 				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
@@ -284,6 +298,20 @@
 					<condition var1="tw_battery" op="&gt;" var2="0"/>
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 					<condition var1="tw_battery_pos_x" op="!=" var2="0"/>
+					<condition var1="tw_remove_battery_text" var2="1"/>
+				</conditions>
+				<font resource="font_m"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
+				<text>%tw_battery%</text>
+			</text>
+
+			<text color="%text_color%">
+				<conditions>
+					<condition var1="tw_no_battery_percent" var2="0"/>
+					<condition var1="tw_battery" op="&gt;" var2="0"/>
+					<condition var1="tw_battery" op="&lt;" var2="101"/>
+					<condition var1="tw_battery_pos_x" op="!=" var2="0"/>
+					<condition var1="tw_remove_battery_text" op="!=" var2="1"/>
 				</conditions>
 				<font resource="font_m"/>
 				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
